library/source/wrapper/DdUrlTransfer.bs

```diff
@@ -624,6 +616,27 @@ class DdUrlTransfer
     sub ClearCookies()
         m.roUrlTransfer.ClearCookies()
     end sub
+
+    sub _traceRequest()
+        m.traceId = generateUniqueId()
+        m.spanId = generateUniqueId()
+        rndTrace = (Rnd(101) - 1) ' Rnd(n) returns a number between 1 and n (both inclusive)
+        ddLogWarning("rndTrace:" + rndTrace.toStr())
+        ddLogWarning("m.tracingSamplingRate:" + m.tracingSamplingRate.toStr())
+        if (rndTrace < m.tracingSamplingRate)
+            ddLogWarning("Network request trace for " + m.GetUrl() + " is KEPT ")
+            m.roUrlTransfer.AddHeader("x-datadog-trace-id", m.traceId)
+            m.roUrlTransfer.AddHeader("x-datadog-parent-id", m.spanId)
+            m.roUrlTransfer.AddHeader("x-datadog-sampling-priority", "1")
+            m.roUrlTransfer.AddHeader("x-datadog-origin", "rum")
+        else
+            ddLogWarning("Network request trace for " + m.GetUrl() + " is DISCARDED  ")
+            m.traceId = invalid
+            m.spanId = invalid
+            m.spanId = generateUniqueId()
```

Comment:
why do we still generate span ID after setting it to `invalid`?

Reply:
That's a typo, nice catch